src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Client.Documents.Session.Loaders;
+using Raven.Client.Documents.Session.Tokens;
+using Raven.Client.Extensions;
+
+namespace Raven.Client.Documents.Session
+{
+    public abstract partial class AbstractDocumentQuery<T, TSelf>
+    {
+        internal List<TimeSeriesIncludesToken> TimeSeriesIncludesTokens;
+
+        protected List<CounterIncludesToken> CounterIncludesTokens;
+
+        internal List<CompareExchangeValueIncludesToken> CompareExchangeValueIncludesTokens;
+
+        /// <summary>
+        /// The paths to include when loading the query
+        /// </summary>
+        protected HashSet<string> DocumentIncludes = new HashSet<string>();
+
+        /// <summary>
+        ///   Includes the specified path in the query, loading the document specified in that path
+        /// </summary>
+        /// <param name = "path">The path.</param>
+        public void Include(string path)
+        {
+            DocumentIncludes.Add(path);
+        }
+
+        /// <summary>
+        ///   Includes the specified path in the query, loading the document specified in that path
+        /// </summary>
+        /// <param name = "path">The path.</param>
+        public void Include(Expression<Func<T, object>> path)
+        {
+            Include(path.ToPropertyPath());
+        }
+
+        public void Include(IncludeBuilder includes)
+        {
+            if (includes == null)
+                return;
+
+            if (includes.DocumentsToInclude != null)
+            {
+                foreach (var doc in includes.DocumentsToInclude)
+                {
+                    DocumentIncludes.Add(doc);
+                }
+            }
+
+            IncludeCounters(includes.Alias, includes.CountersToIncludeBySourcePath);
```

Comment:
Can't `CountersToIncludeBySourcePath` be null?

Reply:
I just moved the old code, but the method 'IncludeCounters' checks that internally.